File: EPI/15.8/swift/15.8_test.swift
import Darwin

extension Pair: Equatable {}

public func == (lhs: Pair, rhs: Pair) -> Bool {
    return lhs.a == rhs.a && lhs.b == rhs.b
}

let tests: [(count: Int, pairs: [Pair])] = [
    (
        count: 0,
        pairs: []
    ),
    (
        count: 1,
        pairs: [
            Pair(a: 0, b: 0),
        ]
    ),
    (
        count: 13,
        pairs: [
            Pair(a: 0, b: 0),
            Pair(a: 1, b: 0),
            Pair(a: 0, b: 1),
            Pair(a: 2, b: 0),
            Pair(a: 1, b: 1),
            Pair(a: 0, b: 2),
            Pair(a: 3, b: 0),
            Pair(a: 2, b: 1),
            Pair(a: 1, b: 2),
            Pair(a: 4, b: 0),
            Pair(a: 0, b: 3),
            Pair(a: 3, b: 1),
            Pair(a: 2, b: 2),
        ]
    ),
]

for test in tests {
    let actual = pairs(count: test.count)
    guard actual == test.pairs else {
        print("For test count \(test.count), expected pairs to be \(test.pairs), but were \(actual)")
        exit(1)
    }
}

File: EPI/Binary Search Trees/Enumerate numbers of the form a + b√2/typescript/15.8_test.ts
import {Composite, smallestComposites} from './15.8';

declare function require(name: string): any;
const assert = require('assert');
const inspect = (v: any) => require('util').inspect(v, {depth: null});

const tests: {count: number, smallestComposites: Composite[]}[] = [
    {
        count: 0,
        smallestComposites: []
    },
    {
        count: 1,
        smallestComposites: [
            {a: 0, b: 0},
        ]
    },
    {
        count: 13,
        smallestComposites: [
            {a: 0, b: 0},
            {a: 1, b: 0},
            {a: 0, b: 1},
            {a: 2, b: 0},
            {a: 1, b: 1},
            {a: 0, b: 2},
            {a: 3, b: 0},
            {a: 2, b: 1},
            {a: 1, b: 2},
            {a: 4, b: 0},
            {a: 0, b: 3},
            {a: 3, b: 1},
            {a: 2, b: 2},
        ]
    },
];

for (const test of tests) {
    const actual = smallestComposites(test.count);
    assert.deepStrictEqual(actual, test.smallestComposites, `For test count ${test.count}, ` +
        `expected smallest composites to be ${inspect(test.smallestComposites)}, ` +
        `but were ${actual}`);
}
